client.go

```diff
@@ -219,7 +243,27 @@ func (s *Client) Run(ctx context.Context,
 		info.Alias, hex.EncodeToString(info.IdentityPubkey[:]),
 	)
 
-	// Setup main context used for cancelation.
+	// Make sure the connected node is a compatible version.
+	version, err := s.lndServices.Versioner.GetVersion(ctx)
+	if err != nil {
+		// The version service has only been added in lnd v0.10.0. If
+		// we get an unimplemented error, it means the lnd version is
+		// definitely older than that.
+		s, ok := status.FromError(err)
+		if ok && s.Code() == codes.Unimplemented {
+			return ErrLndVersionIncompatible
+		}
+		return fmt.Errorf("GetVersion error: %v", err)
+	}
+	err = assertLndVersionCompatible(
```

Comment:
For loop it is also important to check the required sub servers

Reply:
Yes, I forgot about that. Added the check.